src/java/test/org/apache/zookeeper/server/NettyServerCnxnTest.java

```diff
@@ -84,4 +98,78 @@ public void testSendCloseSession() throws Exception {
             zk.close();
         }
     }
+
+    @Test(timeout = 30000)
+    public void testMaxClientConnectionsReached() throws Exception {
+        final int maxClientCnxns = 4;
+        final int numClients = 10;
+        createAndTestConnections(numClients, maxClientCnxns, maxClientCnxns);
+    }
+
+    @Test(timeout = 30000)
+    public void testMaxClientConnectionsDisabled() throws Exception {
+        final int maxClientCnxns = 0; // disabled cnxns limit
+        final int numClients = 10;
+        createAndTestConnections(numClients, maxClientCnxns, numClients);
+    }
+
+    private void createAndTestConnections(int numClients, int maxClientCnxns, int cnxnsAccepted) throws Exception {
+
+        File tmpDir = ClientBase.createTmpDir();
+        final int CLIENT_PORT = PortAssignment.unique();
+
+        ZooKeeperServer zks = new ZooKeeperServer(tmpDir, tmpDir, 3000);
```

Comment:
it seems strange to me to have some tests that start their on zookeeper server and another which uses the one provided by the @before from the superclass. I think we should pick one approach per test class. What do you think?